app/cronjob/trailingTradeIndicator/step/get-trading-view.js

```diff
@@ -212,15 +215,11 @@ const execute = async (funcLogger, rawData) => {
 
   logger.info({ tradingViewRequests }, 'TradingView requests');
 
-  const promises = [];
-
-  _.forIn(tradingViewRequests, async (request, _requestInterval) => {
-    promises.push(
-      retrieveTradingView(logger, request.symbols, request.interval)
-    );
-  });
+  const promises = _.map(tradingViewRequests, request =>
+    retrieveTradingView(logger, request.symbols, request.interval)
+  );
 
-  Promise.all(promises);
+  await Promise.all(promises);
```

Comment:
I can see you've added await on many places including this.
The reason I didn't add await in here is to improve performance. So it does not lock the process.
Especially this line would hold the process very long time depending on the network speed. 

I think what we can do is, rather than await, we can use

```
Promise.all(promises).catch(errror => handleError(err))
```
The `handleError` function will contain the following codes:
https://github.com/chrisleekr/binance-trading-bot/pull/448/files#diff-dfb691bd387c0caaba15a63db9600f2d55aadfe212cbb122b777d1bfee0c7745R41

With this refactoring, we can catch the error as well as improve performance.

This change also applies other await changes.

Reference: https://stackoverflow.com/a/58223461

Reply:
@chrisleekr 
Thanks, that's a good idea!
I tried to catch `cache.hset` error from the `Promise` but it does not help because I `cache.hset` inside loop.
What I did is adding catch on `cache.hset` itself.